modules/dnn/src/net_impl_fuse.cpp

```diff
@@ -662,6 +662,12 @@ void Net::Impl::fuseLayers(const std::vector<LayerPin>& blobsToKeep_)
         if (preferableBackend != DNN_BACKEND_OPENCV && preferableBackend != DNN_BACKEND_CUDA)
             continue;  // Go to the next layer.
 
+        // [TODO] temporarily disabled Concat optimization.
+        // It's not quite compatible with dynamic shapes,
+        // so we need to make sure that we correctly predicted shapes
+        // of all the concatenated tensors and their offsets inside the result
+        // and also properly allocated that concatenated tensor in advance
```

Comment:
File github bug

Reply:
Concat optimization is incompatible with dynamic shapes. It will be disabled anyway in the new engine.